lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -627,25 +646,26 @@ def is_scsi_controller(self, device):
         return isinstance(device, tuple(self.scsi_device_type.values()))
 
     @staticmethod
-    def create_ide_controller():
+    def create_ide_controller(bus_number=0):
         ide_ctl = vim.vm.device.VirtualDeviceSpec()
         ide_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
         ide_ctl.device = vim.vm.device.VirtualIDEController()
         ide_ctl.device.deviceInfo = vim.Description()
         # While creating a new IDE controller, temporary key value
         # should be unique negative integers
         ide_ctl.device.key = -randint(200, 299)
-        ide_ctl.device.busNumber = 0
+        ide_ctl.device.busNumber = bus_number
 
         return ide_ctl
 
     @staticmethod
-    def create_cdrom(ide_ctl, cdrom_type, iso_path=None):
+    def create_cdrom(ide_device, cdrom_type, iso_path=None, unit_number=0):
         cdrom_spec = vim.vm.device.VirtualDeviceSpec()
         cdrom_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
         cdrom_spec.device = vim.vm.device.VirtualCdrom()
-        cdrom_spec.device.controllerKey = ide_ctl.device.key
-        cdrom_spec.device.key = -1
+        cdrom_spec.device.controllerKey = ide_device.key
+        cdrom_spec.device.key = -randint(3000, 3999)
```

Comment:
Ok, this might be an issue if there are duplicate device keys, I'll have a try to see how vCenter handle this scenario. Thank you very much for your reviewing.

Reply:
I had a try on 6.7U2 using this module, 
(1) not set CDROM device key, 
(2) set the same CDROM device keys to CDROM devices,
(3) set device keys to random as above,
all can create VM with CDROMs successfully and with device keys from 3000 to 3003. It seems vCenter not use the specified CDROM key. So I'll keep this and see the result after adding SATA support. Thanks.